stl/inc/optional

```diff
@@ -598,46 +612,118 @@ template <class _Ty>
 optional(_Ty) -> optional<_Ty>;
 
 template <class _Ty1, class _Ty2>
-_NODISCARD constexpr bool operator==(const optional<_Ty1>& _Left, const optional<_Ty2>& _Right) {
-    const bool _Left_has_value = _Left.has_value();
-    return _Left_has_value == _Right.has_value() && (!_Left_has_value || *_Left == *_Right);
+_NODISCARD constexpr bool operator==(const optional<_Ty1>& _Left, const optional<_Ty2>& _Right) noexcept(
+    noexcept(_Fake_copy_init<bool>(*_Left == *_Right))) /* strengthened */
+#ifdef __cpp_lib_concepts
+    requires requires {
+    { *_Left == *_Right } -> _Implicitly_convertible_to<bool>;
+}
+#endif // __cpp_lib_concepts

```

Comment:
No change requested: I asked why constraints are being added because [[optional.relops]](https://eel.is/c++draft/optional.relops#1) has *Mandates*, not *Constraints*. @CaseyCarter explained that this helps a libcxx test pass, and that the Standard probably should be constraining them.